src/nvim/ops.c

```diff
@@ -2602,8 +2602,9 @@ static void op_yank_reg(oparg_T *oap, bool message, yankreg_T *reg, bool append)
 
 static void yank_copy_line(yankreg_T *reg, struct block_def *bd, size_t y_idx)
 {
-  char_u *pnew = xmallocz((size_t)(bd->startspaces + bd->endspaces
-                                   + bd->textlen));
+  assert(bd->startspaces >= 0 && bd->endspaces >= 0 && bd->textlen >= 0);
+  char_u *pnew = xmallocz((size_t)bd->startspaces + (size_t)bd->endspaces
+                          + (size_t)bd->textlen);
```

Comment:
Probably it makes sense to use similar approach as https://github.com/neovim/neovim/pull/10507#discussion_r303278571 : 

```c
int size = bd->startspaces + bd->endspaces + bd->textlen;
assert(size >= 0);
char_u *pnew = xmallocz((size_t)size);

```

Reply:
done